server/index.js

```diff
@@ -17,13 +21,48 @@ module.exports = function Server(config, onStart) {
   // Enable CORS for all routes
   app.use(cors());
 
-  // Start server on conigured hose and port
-  const server = app.listen(config.port, config.host, function listen() {
-    this.rootUrl = `http://${this.address().address}:${this.address().port}`;
-    app.log('serve', `Listening at ${this.rootUrl}`);
-    // Execute callback once server starts
-    if (onStart) onStart.call(this);
-  });
+  // Start server on configured host and port
+  let server;
+
+  if (config.portHttps) {
+    let certFiles;
+    if (!config.httpsKeyPath && !config.httpsCertPath) {
+      certFiles = createCertFiles();
+    } else {
+      certFiles = {
+        key: config.httpsKeyPath,
+        cert: config.httpsCertPath
+      };
+    }
+
+    /* eslint-disable no-sync */
+    const key = fs.readFileSync(certFiles.key);
+    const cert = fs.readFileSync(certFiles.cert);
+    /* eslint-enable no-sync */
+
+    const credentials = {
+      key,
+      cert
+    };
+
+    const httpsServer = https.createServer(credentials, app);
+
+    server = httpsServer.listen(config.portHttps, config.host, function listen(err) {
+      if (err) throw err;
+      this.rootUrl = `https://${this.address().address}:${this.address().port}`;
+      app.log('serve', `Listening at ${this.rootUrl}`);
+      // Execute callback once server starts
+      if (onStart) onStart.call(this);
+    });
+  } else {
+    server = app.listen(config.port, config.host, function listen(err) {
+      if (err) throw err;
+      this.rootUrl = `http://${this.address().address}:${this.address().port}`;
+      app.log('serve', `Listening at ${this.rootUrl}`);
+      // Execute callback once server starts
+      if (onStart) onStart.call(this);
+    });
```

Comment:
There is some redundant behavior between starting the http and https servers, this could be DRYed up later.

Reply:
@ryanricard I've just pushed an update using `lodash.partial` to solve the issue of retaining `this` reference while still passing in a flag to tell the `listen` to log as `http` or `https`.